libs/core/tests/unit_tests/test_tools.py

```diff
@@ -925,6 +930,81 @@ async def foo(bar: str) -> str:
     )
 
 
+@tool
+def foo(bar: Any, bar_config: RunnableConfig) -> Any:
+    """The foo."""
+    assert bar_config["configurable"]["foo"] == "not-bar"
+    assert bar == "baz"
+    return bar
+
+
+@tool
+async def afoo(bar: Any, bar_config: RunnableConfig) -> Any:
+    """The foo."""
+    await asyncio.sleep(0.0001)
+    assert bar_config["configurable"]["foo"] == "not-bar"
+    assert bar == "baz"
+    return bar
+
+
+@tool(infer_schema=False)
```

Comment:
If we're gonna be doing bool toggles should we just define the tool in the test function and then parametrize the args?